cmd/gapir/cc/main.cpp

```diff
@@ -395,8 +536,134 @@ void android_process(struct android_app* app, int32_t cmd) {
 
 }  // namespace
 
+// Extract command line arguments from the extra of Android intent:
+//   adb shell am start -n <...> -e gapir-launch-args "'list of arguments to be
+//   extracted'"
+// Note the quoting: from host terminal adb command, we need to double-escape
+// the extra args string, as it is first quoted by host terminal emulator
+// (e.g. bash), then it must be quoted for the on-device shell.
+void GetArgvFromIntents(struct android_app* state, int* argc, char*** argv) {
+  assert(argc != nullptr);
+  assert(argv != nullptr);
+  assert(*argv == nullptr);
+
+  // The extra flag to indicate arguments
+  const char* intent_flag = "gapir-intent-flag";
+
+  JavaVM* java_vm = state->activity->vm;
+  JNIEnv* jni_env;
+  assert(java_vm->AttachCurrentThread(&jni_env, nullptr) == JNI_OK);
+  jobject activity_clazz = state->activity->clazz;
+  jmethodID get_intent_method_id =
+      jni_env->GetMethodID(jni_env->GetObjectClass(activity_clazz), "getIntent",
+                           "()Landroid/content/Intent;");
+  jobject intent =
+      jni_env->CallObjectMethod(activity_clazz, get_intent_method_id);
+  jmethodID get_string_extra_method_id =
+      jni_env->GetMethodID(jni_env->GetObjectClass(intent), "getStringExtra",
+                           "(Ljava/lang/String;)Ljava/lang/String;");
+  jvalue get_string_extra_args;
+  get_string_extra_args.l = jni_env->NewStringUTF(intent_flag);
+  jstring extra_jstring = static_cast<jstring>(jni_env->CallObjectMethodA(
+      intent, get_string_extra_method_id, &get_string_extra_args));
+
+  std::string extra_string;
+  if (extra_jstring) {
+    const char* extra_cstr = jni_env->GetStringUTFChars(extra_jstring, nullptr);
+    extra_string = extra_cstr;
+    jni_env->ReleaseStringUTFChars(extra_jstring, extra_cstr);
+    jni_env->DeleteLocalRef(extra_jstring);
+  }
+
+  jni_env->DeleteLocalRef(get_string_extra_args.l);
+  jni_env->DeleteLocalRef(intent);
+  java_vm->DetachCurrentThread();
+
+  // Prepare arguments with a value for argv[0], as gflags expects
+  std::vector<std::string> args;
+  args.push_back(
+      "gapir");  // POSIX says the first argv should be the program being run.
+                 // Lets inject a placeholder here for compatibility.
+
+  // Split extra_string
+  std::stringstream ss(extra_string);
+  std::string arg;
+
+  while (std::getline(ss, arg, ' ')) {
```

Comment:
This ignores quoted arguments. I am not sure if we want/need to support that, but things like spaces in filenames etc will break here.

Reply:
I left this behaviour un-altered. The problem you highlight exists in the original code.

Reply:
Sounds good. Feel free to leave this as-is.